File: src/lib/api/type-contracts/system.ts
export interface LLMConfig {
  llm_provider: string;
  llm_fallback_to_mock: boolean;
  openai_base_url: string;
  openai_model: string;
  openai_temperature: number;
  openai_timeout_seconds: number;
  openai_api_key_configured: boolean;
  strategy_openai_base_url?: string;
  strategy_openai_model?: string;
  strategy_openai_timeout_seconds?: number;
  strategy_openai_api_key_configured?: boolean;
}

export interface LLMDryRunResult {
  provider_requested: string;
  provider_used: string;
  fallback_used: boolean;
  raw_preview: string;
  parsed_preview: Record<string, unknown>;
  ok: boolean;
  error?: string | null;
}

export interface ApiHealth {
  status: string;
}

export type ModelRouteStatus = "configured" | "fallback" | "disabled" | "local" | "external";
export type ModelScanStatus = "ready" | "partial" | "blocked";
export type ModelScanRouteStatus = "ready" | "skipped" | "blocked";
export type ModelUpgradeStatus = "applied" | "no_change" | "blocked";

export interface ModelRuntimeRoute {
  key: string;
  label: string;
  provider: string;
  effective_provider: string;
  model?: string | null;
  base_url?: string | null;
  strategy: string;
  fallback: string;
  status: ModelRouteStatus;
  upgrade_managed: boolean;
}

export interface ModuleModelBinding {
  key: string;
  label: string;
  area: string;
  route_key?: string | null;
  provider: string;
  model?: string | null;
  strategy: string;
  fallback: string;
  status: ModelRouteStatus;
  upgrade_managed: boolean;
}

export interface ModelControlPlaneSnapshot {
  generated_at: string;
  policy_version: string;
  routes: ModelRuntimeRoute[];
  modules: ModuleModelBinding[];
}

export interface SupportedModel {
  id: string;
  owned_by: string;
  created?: number | null;
  routes: string[];
  capabilities: string[];
  excluded: boolean;
  exclusion_reason: string;
  scores: Record<string, number>;
  rank_reason: string;
}

export interface ModelScanRoute {
  route_key: string;
  label: string;
  provider: string;
  base_url?: string | null;
  status: ModelScanRouteStatus;
  model_count: number;
  models: string[];
  error_code: string;
  message: string;
}

export interface ModelRecommendation {
  role: "generation" | "strategy" | "vision";
  route_key: string;
  model: string;
  current_model?: string | null;
  change_required: boolean;
  score: number;
  reason: string;
}

export interface SupportedModelScan {
  generated_at: string;
  policy_version: string;
  status: ModelScanStatus;
  total_discovered: number;
  routes: ModelScanRoute[];
  models: SupportedModel[];
  recommendations: ModelRecommendation[];
  message: string;
}

export interface StrongestModelUpgrade {
  generated_at: string;
  status: ModelUpgradeStatus;
  previous_models: Record<string, string | null>;
  applied_models: Record<string, string | null>;
  changed_fields: string[];
  persisted: boolean;
  runtime_reloaded: boolean;
  message: string;
  scan: SupportedModelScan;
}

export type ReleaseReadinessGateStatus = "pass" | "watch" | "blocked";
export type ReleaseReadinessActionPriority = "high" | "medium" | "low";

export interface ReleaseReadinessEvidence {
  label: string;
  status: ReleaseReadinessGateStatus;
  summary: string;
  source: string;
  details: Record<string, unknown>;
}

export interface ReleaseReadinessAction {
  priority: ReleaseReadinessActionPriority;
  owner: string;
  action: string;
  reason: string;
  gate_key: string;
  gate_label: string;
}

export interface ReleaseReadinessOperatorCommand {
  gate_key: string;
  gate_label: string;
  label: string;
  command: string;
  purpose: string;
}

export interface ReleaseReadinessArtifact {
  gate_key: string;
  gate_label: string;
  label: string;
  path: string;
  exists: boolean;
  status: ReleaseReadinessGateStatus;
  summary: string;
}

export interface ReleaseReadinessGate {
  key: string;
  label: string;
  status: ReleaseReadinessGateStatus;
  score: number;
  target: string;
  observed: string;
  summary: string;
  evidence: ReleaseReadinessEvidence[];
  actions: ReleaseReadinessAction[];
}

export interface ReleaseReadinessSnapshot {
  generated_at: string;
  release_version: string;
  overall_status: ReleaseReadinessGateStatus;
  readiness_score: number;
  summary_lines: string[];
  gates: ReleaseReadinessGate[];
  next_actions: ReleaseReadinessAction[];
  operator_commands: ReleaseReadinessOperatorCommand[];
  artifacts: ReleaseReadinessArtifact[];
}

export type InternalSkillStage =
  | "production"
  | "internal_candidate"
  | "third_party_test_package";

export type InternalSkillEvaluationStatus =
  | "passed"
  | "in_progress"
  | "not_evaluated"
  | "blocked";

export type InternalSkillDataBoundary =
  | "local_only"
  | "local_app"
  | "external_optional"
  | "external_blocked";

export type InternalSkillExternalApiStatus =
  | "none"
  | "optional_disabled"
  | "blocked_until_review";

export type InternalSkillSecretStatus =
  | "not_required"
  | "required_for_optional_external_api"
  | "blocked_until_review";

export interface InternalSkillDependency {
  name: string;
  dependency_type: string;
  optional: boolean;
  license: string;
}

export interface InternalSkillRegressionSuite {
  path: string;
  gate: string;
  cadence: string;
}

export interface InternalSkillVersionHistory {
  version: string;
  released_at: string;
  change_summary: string;
  rollback: string;
}

export interface InternalSkillRegistryEntry {
  skill_id: string;
  name: string;
  version: string;
  stage: InternalSkillStage;
  evaluation_status: InternalSkillEvaluationStatus;
  owner: string;
  license: string;
  data_boundary: InternalSkillDataBoundary;
  external_api_status: InternalSkillExternalApiStatus;
  secret_status: InternalSkillSecretStatus;
  default_enabled: boolean;
  default_generation_enabled: boolean;
  admission_reason: string;
  dependencies: InternalSkillDependency[];
  regression_suites: InternalSkillRegressionSuite[];
  applicable_documents: string[];
  baselines: string[];
  version_history: InternalSkillVersionHistory[];
  rollback: string;
  notes: string;
}

export interface InternalSkillGovernanceSummary {
  total_skills: number;
  production_skills: number;
  evaluated_skills: number;
  default_chain_skills: number;
  blocked_from_default_chain: number;
  external_api_skills: number;
  secret_required_skills: number;
  data_egress_modes: string[];
}

export interface InternalSkillRuntimeDiagnostics {
  generated_at: string;
  default_chain_blocking_enforced: boolean;
  unreviewed_default_chain_count: number;
  external_api_status_visible: boolean;
  secret_status_visible: boolean;
  data_egress_status_visible: boolean;
  secret_values_exposed: boolean;
  external_api_skill_ids: string[];
  secret_bound_skill_ids: string[];
  data_egress_modes: string[];
}

export interface InternalSkillGovernanceSnapshot {
  registry_version: string;
  summary: InternalSkillGovernanceSummary;
  diagnostics: InternalSkillRuntimeDiagnostics;
  default_chain_skill_ids: string[];
  blocked_from_default_chain_skill_ids: string[];
  entries: InternalSkillRegistryEntry[];
}

File: backend/app/schemas/system.py
from __future__ import annotations

from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel, Field


ReleaseReadinessGateStatus = Literal["pass", "watch", "blocked"]
ReleaseReadinessActionPriority = Literal["high", "medium", "low"]

InternalSkillStage = Literal["production", "internal_candidate", "third_party_test_package"]
InternalSkillEvaluationStatus = Literal["passed", "in_progress", "not_evaluated", "blocked"]
InternalSkillDataBoundary = Literal["local_only", "local_app", "external_optional", "external_blocked"]
InternalSkillExternalApiStatus = Literal["none", "optional_disabled", "blocked_until_review"]
InternalSkillSecretStatus = Literal["not_required", "required_for_optional_external_api", "blocked_until_review"]
ModelRouteStatus = Literal["configured", "fallback", "disabled", "local", "external"]
ModelScanStatus = Literal["ready", "partial", "blocked"]
ModelScanRouteStatus = Literal["ready", "skipped", "blocked"]
ModelUpgradeStatus = Literal["applied", "no_change", "blocked"]


class ReleaseReadinessEvidenceOut(BaseModel):
    label: str
    status: ReleaseReadinessGateStatus
    summary: str
    source: str = ""
    details: dict[str, Any] = Field(default_factory=dict)


class ReleaseReadinessActionOut(BaseModel):
    priority: ReleaseReadinessActionPriority
    owner: str
    action: str
    reason: str
    gate_key: str = ""
    gate_label: str = ""


class ReleaseReadinessOperatorCommandOut(BaseModel):
    gate_key: str
    gate_label: str
    label: str
    command: str
    purpose: str


class ReleaseReadinessArtifactOut(BaseModel):
    gate_key: str
    gate_label: str
    label: str
    path: str
    exists: bool = False
    status: ReleaseReadinessGateStatus
    summary: str


class ReleaseReadinessGateOut(BaseModel):
    key: str
    label: str
    status: ReleaseReadinessGateStatus
    score: int = 0
    target: str
    observed: str
    summary: str
    evidence: list[ReleaseReadinessEvidenceOut] = Field(default_factory=list)
    actions: list[ReleaseReadinessActionOut] = Field(default_factory=list)


class ReleaseReadinessSnapshotOut(BaseModel):
    generated_at: datetime
    release_version: str
    overall_status: ReleaseReadinessGateStatus
    readiness_score: int = 0
    summary_lines: list[str] = Field(default_factory=list)
    gates: list[ReleaseReadinessGateOut] = Field(default_factory=list)
    next_actions: list[ReleaseReadinessActionOut] = Field(default_factory=list)
    operator_commands: list[ReleaseReadinessOperatorCommandOut] = Field(default_factory=list)
    artifacts: list[ReleaseReadinessArtifactOut] = Field(default_factory=list)


class InternalSkillDependencyOut(BaseModel):
    name: str
    dependency_type: str
    optional: bool = False
    license: str = "internal"


class InternalSkillRegressionSuiteOut(BaseModel):
    path: str
    gate: str
    cadence: str


class InternalSkillVersionHistoryOut(BaseModel):
    version: str
    released_at: str
    change_summary: str
    rollback: str


class InternalSkillRegistryEntryOut(BaseModel):
    skill_id: str
    name: str
    version: str
    stage: InternalSkillStage
    evaluation_status: InternalSkillEvaluationStatus
    owner: str
    license: str
    data_boundary: InternalSkillDataBoundary
    external_api_status: InternalSkillExternalApiStatus
    secret_status: InternalSkillSecretStatus
    default_enabled: bool
    default_generation_enabled: bool
    admission_reason: str
    dependencies: list[InternalSkillDependencyOut] = Field(default_factory=list)
    regression_suites: list[InternalSkillRegressionSuiteOut] = Field(default_factory=list)
    applicable_documents: list[str] = Field(default_factory=list)
    baselines: list[str] = Field(default_factory=list)
    version_history: list[InternalSkillVersionHistoryOut] = Field(default_factory=list)
    rollback: str
    notes: str = ""


class InternalSkillGovernanceSummaryOut(BaseModel):
    total_skills: int
    production_skills: int
    evaluated_skills: int
    default_chain_skills: int
    blocked_from_default_chain: int
    external_api_skills: int
    secret_required_skills: int
    data_egress_modes: list[str] = Field(default_factory=list)


class InternalSkillRuntimeDiagnosticsOut(BaseModel):
    generated_at: str
    default_chain_blocking_enforced: bool
    unreviewed_default_chain_count: int
    external_api_status_visible: bool
    secret_status_visible: bool
    data_egress_status_visible: bool
    secret_values_exposed: bool
    external_api_skill_ids: list[str] = Field(default_factory=list)
    secret_bound_skill_ids: list[str] = Field(default_factory=list)
    data_egress_modes: list[str] = Field(default_factory=list)


class InternalSkillGovernanceSnapshotOut(BaseModel):
    registry_version: str
    summary: InternalSkillGovernanceSummaryOut
    diagnostics: InternalSkillRuntimeDiagnosticsOut
    default_chain_skill_ids: list[str] = Field(default_factory=list)
    blocked_from_default_chain_skill_ids: list[str] = Field(default_factory=list)
    entries: list[InternalSkillRegistryEntryOut] = Field(default_factory=list)


class ModelRuntimeRouteOut(BaseModel):
    key: str
    label: str
    provider: str
    effective_provider: str
    model: str | None = None
    base_url: str | None = None
    strategy: str
    fallback: str
    status: ModelRouteStatus
    upgrade_managed: bool = False


class ModuleModelBindingOut(BaseModel):
    key: str
    label: str
    area: str
    route_key: str | None = None
    provider: str
    model: str | None = None
    strategy: str
    fallback: str
    status: ModelRouteStatus
    upgrade_managed: bool = False


class ModelControlPlaneSnapshotOut(BaseModel):
    generated_at: datetime
    policy_version: str
    routes: list[ModelRuntimeRouteOut] = Field(default_factory=list)
    modules: list[ModuleModelBindingOut] = Field(default_factory=list)


class SupportedModelOut(BaseModel):
    id: str
    owned_by: str = ""
    created: int | None = None
    routes: list[str] = Field(default_factory=list)
    capabilities: list[str] = Field(default_factory=list)
    excluded: bool = False
    exclusion_reason: str = ""
    scores: dict[str, int] = Field(default_factory=dict)
    rank_reason: str = ""


class ModelScanRouteOut(BaseModel):
    route_key: str
    label: str
    provider: str
    base_url: str | None = None
    status: ModelScanRouteStatus
    model_count: int = 0
    models: list[str] = Field(default_factory=list)
    error_code: str = ""
    message: str = ""


class ModelRecommendationOut(BaseModel):
    role: Literal["generation", "strategy", "vision"]
    route_key: str
    model: str
    current_model: str | None = None
    change_required: bool = False
    score: int = 0
    reason: str = ""


class SupportedModelScanOut(BaseModel):
    generated_at: datetime
    policy_version: str
    status: ModelScanStatus
    total_discovered: int = 0
    routes: list[ModelScanRouteOut] = Field(default_factory=list)
    models: list[SupportedModelOut] = Field(default_factory=list)
    recommendations: list[ModelRecommendationOut] = Field(default_factory=list)
    message: str = ""


class StrongestModelUpgradeOut(BaseModel):
    generated_at: datetime
    status: ModelUpgradeStatus
    previous_models: dict[str, str | None] = Field(default_factory=dict)
    applied_models: dict[str, str | None] = Field(default_factory=dict)
    changed_fields: list[str] = Field(default_factory=list)
    persisted: bool = False
    runtime_reloaded: bool = False
    message: str = ""
    scan: SupportedModelScanOut
